### PR TITLE
Major refactors

### PR DESCRIPTION
**Pull request by Juicestus**

### CODE REVIEW

File: base/TorqueStatorSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+package org.texastorque.torquelib.base;
+
+import org.texastorque.torquelib.auto.commands.TorqueExecute;
+
+/**
+ * State based subsystem template.
+ *
+ * @author Justus Languell
+ */
+public abstract class TorqueStatorSubsystem<T extends TorqueState> implements TorqueSubsystem {
```
Comment (0marA) [inline base/TorqueStatorSubsystem.java:16]: You should rename T to something more descriptive

File: base/TorqueStatorSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    protected TorqueStatorSubsystem(final T desiredState, final T lastState) {
+        this.desiredState = desiredState;
+        this.lastState = lastState;
+    }
+
+    public final void setState(final T state) {
+        this.desiredState = state;
+        onStateChange();
```
Comment (0marA) [inline base/TorqueStatorSubsystem.java:31]: It should check if lastState isn't equal to the new State before calling onStateChange.

Comment (Juicestus) [reply base/TorqueStatorSubsystem.java:31]: 👍

File: base/TorqueStatorSubsystem.java (hunk 1 above)
Comment (Juicestus) [reply base/TorqueStatorSubsystem.java:16]: nah https://docs.oracle.com/javase/tutorial/java/generics/types.html

**Pull request merged**